data/registry_model/registry_proxy_model.py

```diff
@@ -131,6 +130,51 @@ def lookup_repository(
             state=repo.state if repo is not None else None,
         )
 
+    def _check_blob_upload_and_prune(
+        self, repo_ref: RepositoryReference, upstream_manifest: ManifestInterface
+    ) -> None:
+        """
+        Checks whether the given manifest fits within the quota size for the namespace
+        the repository is part of. If it doesn't, it prunes older tags in the namespace
+        by marking them expired which is eventually garbage collected by the gc worker.
+
+        Raises QuotaExceededException if the given tag is larger than the max quota
+        allotted for the namespace or if there are not enough tags to prune to free up space.
+        """
+        curr_ns_size = namespacequota.get_namespace_size(repo_ref.namespace_name)
+
+        # if quota limit is not set for the namespace, skip auto pruning of images
+        quotas = namespacequota.get_namespace_quota_list(repo_ref.namespace_name)
+        if quotas:
+            # currently only one quota per namespace is supported
+            max_ns_quota = quotas[0].limit_bytes
+        else:
+            logger.info("No quota configured")
+            return
+
+        # if given manifest_ref is a manifestList, we skip checking quota limit
+        if upstream_manifest.is_manifest_list:
```

Comment:
we probably want to run this check before doing anything in this function. the way it currently is now we're making two queries to the database only to return early when `upstream_manifest` is a manifest list

Reply:
agreed! moved it to the beginning of the func